src/compile/layer.ts

```diff
@@ -120,17 +129,32 @@ export class LayerModel extends Model {
   }
 
   public assembleMarks(): any[] {
-    return assembleLayerSelectionMarks(
+    const marks = assembleLayerSelectionMarks(
       this,
       this.children.flatMap(child => {
         return child.assembleMarks();
       })
     );
+
+    // move label marks to the top
```

Comment:
Can you explain more why we need to move label marks to the top here?

Reply:
If a label mark is under another mark, the labels in the label mark would be behind that mark if they overlap.

Reply:
Also in Vega, a text mark with label transform can only avoid the marks that comes before itself. To be able to avoid the marks that comes after itself, we need to push the text mark to the top.

Reply:
These are great explanations to add to the comment in the code.